scripts/build_test.py

```diff
@@ -689,14 +690,14 @@ def test_generate_app_yaml_with_deploy_mode(self):
         )
 
         app_dev_yaml_temp_file = tempfile.NamedTemporaryFile()
-        app_dev_yaml_temp_file.name = mock_dev_yaml_filepath
+        app_dev_yaml_temp_file.name = mock_dev_yaml_filepath # type: ignore[misc]
```

Comment:
Why is this ignore needed?

Reply:
Actually, I was getting errors like this and
### scripts/build_test.py:775: error: Property "name" defined in "IO" is read-only  [misc]
adding ignores was wrong. I think now I got the reason. As Jukkal told [here](https://github.com/python/typeshed/issues/598#issuecomment-252971714) we were trying to write an object attribute and that was the reason. To remove the error, I have set those values using `setattr`.

Reply:
Hmm, looking at the link it seems that this should be fixed though, maybe you need to use `StringIO` instead of just `IO`?

Reply:
I tried and it is giving error as [it](https://github.com/oppia/oppia/blob/2ebe94cbb8db710751b511d96909cbe67081b5f2/core/utils.py#L98) returns `IO[str]`. Should I stick to `IO` then?

Reply:
Hmm, I think I would need a better explanation why does it not work and maybe what are the relations between `IO` and `StringIO` to understand this.

Reply:
No I mean I would like to know what is the relation between `IO`, `StringIO` (or `BytesIO`) in order to properly assess this error, I would imagine there is some interoperability.

Reply:
Okay, I read https://github.com/python/typeshed/issues/598 again and it seems to me that we need to force the usage of `StringIO` and probably change https://github.com/oppia/oppia/blob/2ebe94cbb8db710751b511d96909cbe67081b5f2/core/utils.py#L103 also to `StringIO` (and https://github.com/oppia/oppia/blob/2ebe94cbb8db710751b511d96909cbe67081b5f2/core/utils.py#L112 to `BytesIO`).

Reply:
done

Reply:
hey, I just want to know why you asked me to change the return type of `open_file` function in `core/utils.py` because I am not getting how it is affecting the `NamedTemporaryFile` object in build_test.py.

Reply:
Ah, you're right. I think the most clean solution is to do setattr with a proper code comment.

Still, can you replace the types in utils.py to be `TextIO` and `BinaryIO`.

Reply:
Sure, I have added some comments and updated the types. PTAL,